app/models/hint_request.rb

```diff
@@ -1,6 +1,6 @@
 class BalanceValidator < ActiveModel::Validator
   def validate(record)
-    if record.team.points < 0
+    if record.team.points < record.bounty
```

Comment:
When the `validate` method is reached, the models have already been updated with the new values. Only when the validations fail, a rollback is performed and the old values restored.

So, at this point, `record.team.points` actually refers to the new balance, after the bounty has been deduced, and we only need this number to be non-negative.